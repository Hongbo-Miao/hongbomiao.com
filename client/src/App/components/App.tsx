import React, { lazy } from 'react';
import { BrowserRouter, Redirect, Route, Switch } from 'react-router-dom';
import HmLazyComponent from '../../shared/components/LazyComponent';
import Paths from '../../shared/utils/paths';
import styles from './App.module.css';

const HmHome = lazy(() => import('../../Home/components/Home'));
const HmFooter = lazy(() => import('./Footer'));

const App: React.FC = () => (
  <div className={styles.hmApp}>
    <BrowserRouter>
      <HmLazyComponent>
        <Switch>
          <Route exact path={Paths.appRootPath} component={HmHome} />
          <Redirect to={Paths.appRootPath} />
        </Switch>
      </HmLazyComponent>
    </BrowserRouter>

    <HmLazyComponent>
      <HmFooter />
    </HmLazyComponent>
  </div>
);

export default App;
