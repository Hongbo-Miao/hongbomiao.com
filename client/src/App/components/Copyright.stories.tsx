import { storiesOf } from '@storybook/react';
import React from 'react';

import HmCopyright from './Copyright';

storiesOf('Copyright', module).add('default', () => <HmCopyright year={1990} />);
