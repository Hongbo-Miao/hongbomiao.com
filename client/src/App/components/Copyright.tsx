import React from 'react';
import styles from './Copyright.module.css';

interface Props {
  year: number;
}

const Copyright: React.FC<Props> = (props: Props) => {
  const { year } = props;
  const copyright = `Â© ${year} H.M.`;
  return <div className={styles.hmCopyright}>{copyright}</div>;
};

export default Copyright;
