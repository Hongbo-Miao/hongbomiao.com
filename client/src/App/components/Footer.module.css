.hmFooter {
  background-color: transparent;
  padding: 64px 80px;
}

.hmContainer {
  max-width: 640px;
}
