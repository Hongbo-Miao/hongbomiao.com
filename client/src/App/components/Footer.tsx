import React, { lazy } from 'react';

import HmLazyComponent from '../../shared/components/LazyComponent';
import styles from './Footer.module.css';
import WEBSITES from '../fixtures/websites';

const HmCopyright = lazy(() => import('./Copyright'));
const HmSocialList = lazy(() => import('./SocialList'));

const Footer: React.FC = () => {
  const year = new Date().getFullYear();

  return (
    <footer className={`footer ${styles.hmFooter}`}>
      <div className={`container ${styles.hmContainer}`}>
        <nav className="level">
          <div className="level-left">
            <HmLazyComponent>
              <HmSocialList websites={WEBSITES} />
            </HmLazyComponent>
          </div>
          <div className="level-right">
            <div className="level-item">
              <HmLazyComponent>
                <HmCopyright year={year} />
              </HmLazyComponent>
            </div>
          </div>
        </nav>
      </div>
    </footer>
  );
};

export default Footer;
