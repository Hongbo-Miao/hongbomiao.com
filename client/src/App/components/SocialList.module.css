.hmLogo {
  transition: all 0.3s ease;
}

.hmLogo:hover {
  opacity: 0.7;
}
