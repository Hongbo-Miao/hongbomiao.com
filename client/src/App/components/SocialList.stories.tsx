import React from 'react';
import { storiesOf } from '@storybook/react';

import HmSocialList from './SocialList';
import WEBSITES from '../fixtures/websites';

storiesOf('SocialList', module)
  .add('default', () => <HmSocialList websites={WEBSITES} />)
  .add('empty', () => <HmSocialList websites={[]} />);
