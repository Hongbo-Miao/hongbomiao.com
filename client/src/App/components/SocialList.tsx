import React from 'react';
import Website from '../types/Website.type';
import styles from './SocialList.module.css';

interface Props {
  websites: ReadonlyArray<Website>;
}

const SocialList: React.FC<Props> = (props) => {
  const { websites } = props;

  const socialListItems = websites.map((website) => {
    const { name, src, url } = website;

    return (
      <div key={name} className="level-item">
        <a href={url} target="_blank" rel="noopener noreferrer">
          <img className={styles.hmLogo} src={src} height="28" width="28" alt={name} />
        </a>
      </div>
    );
  });

  return <>{socialListItems}</>;
};

export default SocialList;
