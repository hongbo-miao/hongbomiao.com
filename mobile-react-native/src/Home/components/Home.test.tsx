import * as eva from '@eva-design/eva';
import { render } from '@testing-library/react-native';
import { ApplicationProvider } from '@ui-kitten/components';
import React from 'react';
import HmHome from './Home';

describe('<HmHome />', () => {
  test('Home', () => {
    const { getByTestId, toJSON } = render(
      // eslint-disable-next-line react/jsx-props-no-spreading
      <ApplicationProvider {...eva} theme={eva.light}>
        <HmHome />
      </ApplicationProvider>,
    );
    expect(getByTestId('bio').props.children).toBe('Making magic happen');
    expect(toJSON()).toMatchSnapshot();
  });
});
