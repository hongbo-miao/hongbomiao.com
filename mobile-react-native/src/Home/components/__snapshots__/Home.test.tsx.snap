// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HmHome /> Home 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    appearance="default"
    level="1"
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
        },
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      appearance="default"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "fontFamily": "NeoSansProBold",
            "fontSize": 35,
            "marginTop": 20,
            "textTransform": "uppercase",
          },
        ]
      }
      testID="name"
    >
      Hongbo Miao
    </Text>
    <Text
      appearance="default"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "fontFamily": "NeoSansProRegular",
            "fontSize": 20,
            "marginTop": 40,
          },
        ]
      }
      testID="bio"
    >
      Making magic happen
    </Text>
  </View>
</View>
`;
